File: src/plt/main/figure_4b.R
# ######################################################
# figure_4b.R 
#
# Volcano plots of snoRNAs in control vs Dicer-knockdown
#
# amadis.pages@upf.edu
# amadis.pages@crg.eu
# ######################################################

# Load libraries
library(ggplot2)
library(RSvgDevice)

# Read command line arguments
command_args <- commandArgs(trailingOnly=TRUE)
stats <- command_args[1]
svgof <- command_args[2]
cutoff <- as.numeric(command_args[3])

# Load data
dat <- read.table(stats, sep = "\t")
colnames(dat) <- c("type", "annotation", "logfold", "padj", "Significant")

# Plot figure
devSVG(file = svgof, bg = "white", fg = "black", onefile = TRUE, xmlHeader = TRUE)
ggplot(dat[dat$type == "snoRNA",], aes(x = logfold, y = -log(padj, 10), col = Significant)) + geom_point() + facet_grid(annotation ~ ., scale = "free_y") + geom_hline(yintercept = -log(cutoff,10), size =.1)
dev.off()

File: src/plt/main/figure_4b.pl
# #####################################################
# figure_4b.pl
#
# Volcano plot of snoRNAs in control vs Dicer-knockdown
#
# amadis.pages@upf.edu
# amadis.pages@crg.eu
# #####################################################

use strict;

# Usage:
#   perl figure_4b.pl deseq_file cutoff
# Example
#   perl figure_4b.pl dat/dicer/serpent_annotation_extended.deseq > tmp/inputR.tab

# Parse command line
my ($serpent_a_f, $cutoff) = @ARGV;

# Parse file
open(SER, " < $serpent_a_f");
while(<SER>) {
  chomp;
  next if (m/baseMean/);
  
  my ($id, $baseMean, $log2FoldChange, $lfcSE, $stat, $pvalue, $padj) = split "\t";
  next if ($padj eq "NA");
  
  my ($chr, $sbp, $ebp, $bio, $ano) = split "#", $id;
  my $sig = "No";
  $sig = "Yes" if ($padj < $cutoff);
  my $ori = "Gencode";
  $ori = "SeRPeNT" if ($ano >= 0);

  print "$bio\t$ori\t$log2FoldChange\t$padj\t$sig\n";
}
close(SER);
